app/abilities/lecture_ability.rb

```diff
@@ -33,7 +33,7 @@ def initialize(user)
     end
 
     can :subscribe_page, Lecture do |lecture|
-      lecture.published? || !user.generic?
+      lecture.published? || user.active_teachable_editor?
```

Comment:
Why can only active teachable editors subscribe to a lecture?

Reply:
ah I see now, you're shortcircuiting here ;) If a lecture is unpublished, then only users with the respective rights can subscribe to it. Ok, so this is fine.